### PR TITLE
BME-115 Fix keybinds not being configurable

### PR DESCRIPTION
**Pull request by maddie-j**

### CODE REVIEW

File: src/main/java/com/eerussianguy/blazemap/feature/BlazeMapFeaturesClient.java [hunk 1]
```diff
@@ -23,19 +23,18 @@
...
 public class BlazeMapFeaturesClient {
-    public static final KeyMapping KEY_MAPS = new KeyMapping("blazemap.key.maps", KeyConflictContext.IN_GAME, InputConstants.Type.KEYSYM, GLFW.GLFW_KEY_B, BlazeMap.MOD_NAME);
-    public static final KeyMapping KEY_ZOOM = new KeyMapping("blazemap.key.zoom", KeyConflictContext.IN_GAME, InputConstants.Type.KEYSYM, GLFW.GLFW_KEY_LEFT_BRACKET, BlazeMap.MOD_NAME);
-    public static final KeyMapping KEY_WAYPOINTS = new KeyMapping("blazemap.key.waypoints", KeyConflictContext.IN_GAME, InputConstants.Type.KEYSYM, GLFW.GLFW_KEY_N, BlazeMap.MOD_NAME);
+    public static final Lazy<KeyMapping> KEY_MAPS = Lazy.of(() -> new KeyMapping("blazemap.key.maps", KeyConflictContext.IN_GAME, InputConstants.Type.KEYSYM, GLFW.GLFW_KEY_B, BlazeMap.MOD_NAME));
+    public static final Lazy<KeyMapping> KEY_ZOOM = Lazy.of(() -> new KeyMapping("blazemap.key.zoom", KeyConflictContext.IN_GAME, InputConstants.Type.KEYSYM, GLFW.GLFW_KEY_LEFT_BRACKET, BlazeMap.MOD_NAME));
+    public static final Lazy<KeyMapping> KEY_WAYPOINTS = Lazy.of(() -> new KeyMapping("blazemap.key.waypoints", KeyConflictContext.IN_GAME, InputConstants.Type.KEYSYM, GLFW.GLFW_KEY_N, BlazeMap.MOD_NAME));
```
Comment (maddie-j) [inline src/main/java/com/eerussianguy/blazemap/feature/BlazeMapFeaturesClient.java:28]: The Forge docs wanted this `Lazy<>` thing so I added it 🤷‍♀️

File: src/main/java/com/eerussianguy/blazemap/FMLEventHandler.java [hunk 2]
```diff
@@ -11,6 +12,7 @@ public class FMLEventHandler {
     public static void init() {
         final IEventBus bus = FMLJavaModLoadingContext.get().getModEventBus();
 
+        bus.addListener(BlazeMapFeaturesClient::onKeyBindRegister);
```
Comment (maddie-j) [inline src/main/java/com/eerussianguy/blazemap/FMLEventHandler.java:15]: I had thought the `@SubscribeEvent` had registered the event automatically, but turns out I needed to manually register it for it to fire and therefore have the key control config menu recognise it.

Comment (Mrthomas20121) [reply src/main/java/com/eerussianguy/blazemap/FMLEventHandler.java:15]: You could've also done it this way with the annotation `@Mod.EventBusHandler` on the `BlazeMapFeaturesClient` class tbf

Comment (LordFokas) [reply src/main/java/com/eerussianguy/blazemap/FMLEventHandler.java:15]: This can stay for now, later we'll review all the event stuff in 1.20

**Pull request merged**